### ISSUE TITLE
[Feature] Self-repair from another client

### ISSUE THREAD

**Issue by LoganDark** [BetterDiscord/BetterDiscord#1641]
### Before Requesting

- [X] I found no existing issue matching my feature request

### Describe the feature you'd like!

TL;DR: Allow using the BD auto-updater to repair another installation of BD. 

Imagine you have two Discord clients, say, stable and PTB. You have Stable running, and it is a functioning BD installation, complete with BD's auto-updater.

You start PTB, and it installs 14 updates, and BetterDiscord is gone. But you still have Stable running with its own BD.

What if you could use the Stable version's auto-updater (the "Updates" section in settings) to install BD onto PTB?

If you're familiar with Android rooting on A/B devices, this is basically the "install to inactive partition" feature of Magisk: use the current runtime environment (Stable with functioning BD) in order to patch the one you're about to boot into (PTB).

And then I could do the same in reverse: once PTB is working, I can restart+update Stable, and then use PTB to reinstall BD onto Stable.

I think this would be a good failsafe for cases where BD doesn't automatically persist itself.

**Comment by rauenzi** [comment]
> If you're familiar with Android rooting on A/B devices, this is basically the "install to inactive partition" feature of Magisk: use the current runtime environment (Stable with functioning BD) in order to patch the one you're about to boot into (PTB).

This environment is very different from Magisk/Android, and I think the vast majority of users are not even using multiple release channels.

Having an injector inside of the BD client itself feels out of scope.

**Comment by LoganDark** [comment]
> This environment is very different from Magisk/Android

I feel like you misunderstood the point of the example. I am not proposing an A/B partitioning scheme. I'm proposing using a working client to repair a broken one.

I thought BD gained an injector already when you added the auto-updater?

**Comment by rauenzi** [comment]
> I feel like you misunderstood the point of the example. I am not proposing an A/B partitioning scheme. I'm proposing using a working client to repair a broken one.

I'm just stating that they are conceptually different so even though that concept of fixing one from another makes sense there doesn't mean it makes sense here.

> I thought BD gained an injector already when you added the auto-updater?

Nope, it only updates.

**Comment by LoganDark** [comment]
> even though that concept of fixing one from another makes sense there doesn't mean it makes sense here.

Maybe you could at least make the installer/injector auto-update so it doesn't need to be manually redownloaded?

By the time my BD install breaks, the installer on my computer's usually pretty outdated. So I typically have to go to GitHub and manually download a new installer immediately if I want BD again.

Repairing from another client was my idea since I had another one open at the time, but probably most any way of helping this flow would be much appreciated.

**Comment by rauenzi** [comment]
The installer itself already has an update check and will download the newest version for you. The installer doesn't update very frequently either

**Comment by LoganDark** [comment]
OK, sorry. (It didn't work for me last time which prompted this issue, it just opened the GitHub releases page in my browser. but maybe my installer was just that old??)

**Comment by xdCoolesding** [comment]
You could also have an auto updater that the starts discord, so you don't start discord directly, but you start a script that first checks if an update is available for BD and if not or it's done it just starts discord. Probably a big security risk though

**Comment by LoganDark** [comment]
> You could also have an auto updater that the starts discord, so you don't start discord directly, but you start a script that first checks if an update is available for BD and if not or it's done it just starts discord. Probably a big security risk though

No, you'd need to apply the latest discord updates *before* applying the latest BD updates. The current model is the best they can do.
